c++/src/kj/compat/http.c++

```diff
@@ -2667,6 +2894,34 @@ private:
 
     Mask mask(maskKeyGenerator);
 
+    bool useCompression = false;
+    kj::Maybe<kj::Array<byte>> compressedMessage;
```

Comment:
You need to make sure this gets `.attach()`ed to the write promise below (just like `ownMessage`), otherwise if the message doesn't get fully written to the socket buffer on the first syscall, it'll be used after free on subrequest writes.

Reply:
https://github.com/capnproto/capnproto/pull/1505/commits/bff88eec0bc71fb2270d5b632d4ce2c21527a8cf